### PR TITLE
noteTools added for POST and PATCH /note

### PR DESCRIPTION
**Pull request by e11sy**
## Problem
```
noteTools: {
  name: string,
  id: string
}
```
now noteTools are not send to api on POST and PATCH /note, but they are required in body, that is why API returns 500

## Solution
- added noteTools entity and noteTools formalizatoin for useNote service
- added noteTools, that are formed from content blocks ( via user tools )
- now on POST and PATCH /note `tools: noteTools` are included to body

### CODE REVIEW

File: src/application/services/useNote.ts [hunk 1]
```diff
@@ -166,15 +167,24 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
...
+    const { userEditorTools } = useAppState();
+
     if (note.value === null) {
       throw new Error('Note is not loaded yet');
     }
 
+    /**
+     * Format specified tools
+     */
+    const specifiedNoteTools = content.blocks.map((block) => {
```
Comment (elizachi) [inline src/application/services/useNote.ts:179]: What means specified note tools?

**Review by elizachi — APPROVED** [review]

File: src/application/services/useNote.ts [hunk 2]
```diff
@@ -166,15 +167,24 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
...
+      return { name: block.type, id: (userEditorTools.value ?? []).find((tool) => tool.name === block.type)!.id };
```
Comment (neSpecc) [inline src/application/services/useNote.ts:180]: lets make the `resolveToolByName()` method

File: src/domain/entities/Note.ts [hunk 3]
```diff
@@ -10,6 +11,18 @@ export type NoteId = string;
  */
 export type NoteContent = OutputData;
 
+export type NoteTools = {
```
Comment (neSpecc) [inline src/domain/entities/Note.ts:14]: add entity description

File: src/application/services/useNote.ts (hunk 2 above)
Comment (TatianaFomina) [inline src/application/services/useNote.ts:180]: note can contain tools that are not installed by the user. For example, there are 2 collaborators on the note. One has list plugin installed, the other does not. If the second user tries to update the note, what will happen?

File: src/application/services/useNote.ts (hunk 1 above)
Comment (e11sy) [reply src/application/services/useNote.ts]: these are tools, that would be specified in the api request

File: src/application/services/useNote.ts [hunk 4]
```diff
@@ -159,6 +166,19 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
...
+  /**
+   * Returns list of tools used in the note
+   *
+   * @param content - content of the note
+   */
+  async function resolveToolsByContent(content: NoteContent): Promise<NoteTool[]> {
+    const { allTools } = useTools(noteTools);
+
+    return content.blocks.map((block) => {
+      return { name: block.type, id: (allTools.value ?? []).find((tool) => tool.name === block.type)!.id };
```
Comment (neSpecc) [inline src/application/services/useNote.ts:178]: its hard to read, please, write it via the separate statements

File: src/application/services/useNote.ts [hunk 5]
```diff
@@ -159,6 +166,19 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
...
 
+  /**
+   * Returns list of tools used in the note
+   *
+   * @param content - content of the note
+   */
+  async function resolveToolsByContent(content: NoteContent): Promise<NoteTool[]> {
+    const { allTools } = useTools(noteTools);
+
+    return content.blocks.map((block) => {
```
Comment (neSpecc) [inline src/application/services/useNote.ts:177]: map won't be unique

File: src/application/services/useTools.ts [hunk 6]
```diff
@@ -63,20 +65,21 @@ export function useTools(noteTools: Ref<EditorTool[]>): {
...
       }
-      tools.value = await downloadTools(allTools);
+      tools.value = await downloadTools(allTools.value);
     },
     { immediate: true }
   );
 
   return {
     tools,
+    allTools,
```
Comment (neSpecc) [inline src/application/services/useTools.ts:83]: There is a problem with naming: 
seems like `tools` and `allTools` contains the same set. But first contains downloaded something, and all tools contains just info. 

I'w suggest:

`tools` -> `toolsConnected`
`allTools` -> `tools`

File: src/application/services/useNote.ts (hunk 4 above)
Comment (TatianaFomina) [reply src/application/services/useNote.ts:178]: ➕

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**